ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStreamWriter.java

```diff
@@ -41,62 +40,135 @@
 import java.io.EOFException;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.io.StringWriter;
 import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import static io.confluent.ksql.rest.server.resources.streaming.TopicStreamWriter.Format.getFormatter;
+
 
 public class TopicStreamWriter implements StreamingOutput {
+  public enum Format {
+
+    UNDEFINED {
+  },
+    AVRO {
+      private String topicName;
+      private KafkaAvroDeserializer avroDeserializer;
+
+      @Override
+      public boolean isFormat(String topicName, ConsumerRecord<String, Bytes> record,
+          SchemaRegistryClient schemaRegistryClient) {
+        this.topicName = topicName;
+        try {
+          avroDeserializer = new KafkaAvroDeserializer(schemaRegistryClient);
+          avroDeserializer.deserialize(topicName, record.value().get());
+          return true;
+        } catch (Throwable t) {
+          return false;
+        }
+      }
+
+      @Override
+      String print(ConsumerRecord<String, Bytes> consumerRecord) throws IOException {
+        String time = dateFormat.format(new Date(consumerRecord.timestamp()));
+        GenericRecord record = (GenericRecord) avroDeserializer.deserialize(topicName, consumerRecord.value().get());
+        String key = consumerRecord.key() != null ? consumerRecord.key() : "null";
+        return new StringBuilder().append(time).append(", ").append(key).append(", ")
+          .append(record.toString()).append("\n").toString();
+      }
+    },
+    JSON {
+      ObjectMapper objectMapper = new ObjectMapper();
+      @Override
+      public boolean isFormat(String topicName, ConsumerRecord<String, Bytes> record,
+          SchemaRegistryClient schemaRegistryClient) {
+        try {
+          objectMapper.readTree(record.value().toString());
+          return true;
+        } catch (Throwable t) {
+          return false;
+        }
+      }
+
+      @Override
+      String print(ConsumerRecord<String, Bytes> record) throws IOException {
+        JsonNode jsonNode = objectMapper.readTree(record.value().toString());
+        ObjectNode objectNode = objectMapper.createObjectNode();
+        objectNode.put(SchemaUtil.ROWTIME_NAME, record.timestamp());
+        objectNode.put(SchemaUtil.ROWKEY_NAME, (record.key() != null)? record.key().toString(): "null");
+        objectNode.setAll((ObjectNode) jsonNode);
+        StringWriter stringWriter = new StringWriter();
+        objectMapper.writeValue(stringWriter, objectNode);
+        return stringWriter.toString() + "\n";
+      }
+    },
+    DELIMITED {
```

Comment:
maybe just call this DEFAULT ? Since it works for any format that we don't explicitly handle.

Reply:
good point - i changed it to string, because it presumed the deserialized data can be stringified